src/forwarddiff.jl

```diff
@@ -0,0 +1,73 @@
+using .ForwardDiff: ForwardDiff, DiffResults, StaticArrays
+
+struct ForwardDiffBackend <: AbstractBackend end
+
+@primitive function pushforward_function(ba::ForwardDiffBackend, f, xs...)
+    return function pushforward(vs)
+        return ForwardDiff.derivative(h -> f(step_toward.(xs, vs, h)...), 0)
+    end
+end
+function pushforward_function(::ForwardDiffBackend, f, x)
+    return function pushforward(v)
+        if v isa Tuple
+            @assert length(v) == 1
+            return (ForwardDiff.derivative(h -> f(step_toward(x, v[1], h)), 0),)
+        else
+            return (ForwardDiffFD.derivative(h -> f(step_toward(x, v, h)), 0),)
+        end
+    end
+end
+
+primal_value(::ForwardDiffBackend, ::Any, f, xs) = ForwardDiff.value.(f(xs...))
```

Comment:
```suggestion
primal_value(x::ForwardDiff.Dual) = ForwardDiff.value(x)
```

Reply:
With this, all the `value_and_xx` will be much faster by default. We should change this in the tests too.

Reply:
Please update the benchmarks. I think we may not need to define the value_and methods at all. We probably need the others though. So we can start with the Jacobian primitive, then define the methods without value, and check if the value methods are needed.